### PR TITLE
core: speed up header import

### PR DESCRIPTION
**Pull request by holiman**
This is a follow up to https://github.com/ethereum/go-ethereum/pull/21471. 

Charts and niceties can be found here: https://github.com/holiman/headerimport/blob/main/README.md

This PR implements the following modifications
- [x] 1M blocks without header-shortcut (mod1)
  - Do not shortcut check if block is present, thus avoid disk lookup
- [x] 1M blocks with less hashing (mod2)
  - Don't check hash ancestry in early-check (it's still done in parallel checker)
- [ ] 1M blocks with less syncronization (mod3)
  - Don't check for abort between every header EDIT: this optimization was reverted
- [x] 1M blocks without `Now`
  - Don't check `time.Now` for every single header

TLDR: With these changes, writing 1M headers goes down to `62s` for 1M headers, from `106s` on (old) master, and `80s` on PR 21471.

### CODE REVIEW

File: core/rawdb/freezer_table.go [hunk 1]
```diff
@@ -103,6 +103,11 @@ type freezerTable struct {
 	lock   sync.RWMutex // Mutex protecting the data file descriptors
 }
 
+// NewFreezerTable opens the given path as a freezer table.
```
Comment (MariusVanDerWijden) [inline core/rawdb/freezer_table.go:106]: ?

File: core/headerchain.go [hunk 2]
```diff
@@ -338,16 +339,16 @@ func (hc *HeaderChain) ValidateHeaderChain(chain []*types.Header, checkFreq int)
 
 	// Iterate over the headers and ensure they all check out
 	for i := range chain {
-		// If the chain is terminating, stop processing blocks
-		if hc.procInterrupt() {
-			log.Debug("Premature abort during headers verification")
-			return 0, errors.New("aborted")
-		}
 		// Otherwise wait for headers checks and ensure they pass
```
Comment (MariusVanDerWijden) [inline core/headerchain.go:342]: This comment doesn't make sense anymore

File: core/rawdb/freezer_table.go (hunk 1 above)
Comment (holiman) [reply core/rawdb/freezer_table.go:106]: ... What is the question?
If you're wondering why I exposed this, it's to be able to run the header-import via https://github.com/holiman/headerimport

File: core/headerchain.go (hunk 2 above)
Comment (holiman) [inline core/headerchain.go:342]:
```suggestion
		// Wait for headers checks and ensure they pass
```

File: core/rawdb/freezer_table.go (hunk 1 above)
Comment (MariusVanDerWijden) [reply core/rawdb/freezer_table.go:106]: Ahh okay, I wondered because I didn't see it used in the code and thought it might be an artifact

File: consensus/ethash/consensus.go [hunk 3]
```diff
@@ -180,10 +181,7 @@ func (ethash *Ethash) verifyHeaderWorker(chain consensus.ChainHeaderReader, head
 	if parent == nil {
 		return consensus.ErrUnknownAncestor
 	}
-	if chain.GetHeader(headers[index].Hash(), headers[index].Number.Uint64()) != nil {
```
Comment (ucwong) [inline consensus/ethash/consensus.go:183]: Why remove this check ?

Comment (holiman) [reply consensus/ethash/consensus.go:183]: See https://github.com/holiman/headerimport/blob/main/README.md#1m-blocks-on-21471

**Comment by holiman** [comment]
Rebased on master. Master: 
total-times_master-2021-01-11_1M

This PR: 
total-times_21967-2021-01-11_1M

TLDR: `52+22 = 74s` vs `38+23=61s`

File: consensus/ethash/consensus.go [hunk 4]
```diff
@@ -130,11 +130,12 @@ func (ethash *Ethash) VerifyHeaders(chain consensus.ChainHeaderReader, headers [
 		done   = make(chan int, workers)
 		errors = make([]error, len(headers))
 		abort  = make(chan struct{})
+		utcNow = time.Now().Unix()
```
Comment (karalabe) [inline consensus/ethash/consensus.go:133]: unix :)

**Review by karalabe — APPROVED** [review]
Pls fix utx->unix, othewise SGTM
